### PR TITLE
rename MinimumOSPlatformAttribute -> SupportedOSPlatformAttribute

### PR DESCRIPTION
**Pull request by adamsitnik**
@wli3 this is the follow up of https://github.com/dotnet/runtime/pull/40371

I was not sure whether I should also rename the "MinimumOSPlatform" property to "SupportedOSPlatform". Any thoughts on that?

### CODE REVIEW

**Comment by wli3** [comment]
Yes. We do need to change the property. Could help double check it did not use anywhere else?

**Comment by adamsitnik** [comment]
> Could help double check it did not use anywhere else?

@wli3 done! please let me know if anything else needs to be changed. Thanks!

**Comment by wli3** [comment]
The next runtime insertion with the type change will be blocked the failure tests. And by when we need to cherry-pick these changes to the runtime insertion PR to unblock it.

**Review by wli3 — APPROVED** [review]

**Comment by wli3** [comment]
For some reason it did not work. The new runtime is in but, it will fail to build

**Comment by adamsitnik** [comment]
> For some reason it did not work. The new runtime is in but, it still fail to build

could you please point me to the logs or paste the full error message here?

**Comment by wli3** [comment]
@adamsitnik the current master should have the type rename flow in already. The fact that this PR still fail to build is the problem.

It should have something to do with how SDK build and test. But I haven't get time to look into it yet.

**Comment by adamsitnik** [comment]
> It should have something to do with how SDK build and test.

I've synced my branch and was able to reproduce the problem locally.

For some reason, there are two .NET 5 versions being downloaded:
* 5.0.0-rc.1.20371.13 (uses the old name) 
* 5.0.0-rc.1.20410.10 (uses the new name)

The question is: how can we force the tooling to use the newer version?

**Comment by wli3** [comment]
It is not really what SDK to install. Since the new installed SDK won't have the type change (get stuck here https://github.com/dotnet/installer/pull/8129). During SDK test, it should use the latest runtime and reference assemblies of the new runtime. But for some reason it doesn't hence the missing type error.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets [hunk 1]
```diff
@@ -117,11 +117,11 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
 
     <ItemGroup Condition="'$(TargetPlatformIdentifier)' != ''
-                         and '$(MinimumOSPlatform)' != ''
```
Comment (pranavkm) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets:120]: @adamsitnik \ @wli3  do you happen to know what values I should use to make this work for Blazor? What values do I need to specify in the SDK for `TargetPlatformIdentifier` and `MinimumOSPlatform`?

**Comment by adamsitnik** [comment]
> During SDK test, it should use the latest runtime and reference assemblies of the new runtime. But for some reason it doesn't hence the missing type error.

FWIW I've taken a look at the problem and the `dotnet` process that is running the unit tests itself (the host process) is using the latest bits (`5.0.0-rc.1.20410.10`) and knows what `SupportedOSPlatformAttribute` type is.

The `dotnet` process  spawned by the tests to build the generated test projects (`sdk\artifacts\bin\redist\Release\dotnet\dotnet.exe run` in my example) is using the old bits (`5.0.0-rc.1.20371.13`) and it fails to compile the project generated by the tests.

The `sdk\artifacts\bin\redist\Release\dotnet\shared\Microsoft.NETCore.App` contains `5.0.0-rc.1.20410.10` folder, so whichever setting tells dotnet host proces to use `Microsoft.NETCore.App=5.0.0-rc.1.20410.10` is not applied to the dotnet process spawned by the tests.

**Comment by wli3** [comment]
@adamsitnik thank you! that's great investigation. Sorry I've fire fighting the preview 8 release

**Comment by wli3** [comment]
I think I have a plan. The BundledNETCoreAppPackageVersion still uses the stage 0 runtime. So it builds against the last version. I plan to just override the version with XML in C# at https://github.com/dotnet/sdk/blob/b1223209644d900702287faea8e9b71f95ec49f8/src/Layout/redist/targets/OverlaySdkOnLKG.targets#L15

Use a task to override since it was generated as a string literal replace anyway. And using C# can have better error when anything goes wrong. cc @dsplaisted

**Comment by jeffhandley** [comment]
> Merged #12775 into master.
Great work, @wli3 and @adamsitnik!

**Comment by adamsitnik** [comment]
@wli3 thanks a lot!

**Pull request merged**